### DISCUSSION TITLE
Datasync PushFailed

### DISCUSSION THREAD

**Discussion by geath** [Azure/azure-mobile-apps#500] [Q&A]
Hello, 
I'm playing around with the demo ToDoApp and I'm experiencing the following issue.  When PushItemsAsync is called from my offline table the following exceptions occur : 
 - OfflineStoreException: Failed to read errors from the local store.
- JsonReaderException: Error reading JObject from JsonReader. Path '', line 0, position 0.

However,  despite the error, PushItemsAsync  updates the remote table.   

Any ideas?

**Comment by adrianhall** [comment]
Nothing that I can determine just from this information.  What platform are you using, for starters?  Which step are you on?  What does your Initialize method look like after you made the adjustments?  What about your Synchronize method?

**Reply by geath** [reply]
Dear Adrian thanks for your reply!

The notion is to test the functionality of pushing entries existing only in the offline store (SQLite) to remote table. I reproduced successfully the steps to simulate the lack of network connection and I managed to use the offline store.  
After restoring the connection,  I press the refresh button in UI (MainPage)  which  is bind with the RefreshItemsAsync method in RemoteTodoService class .

From there a  subsequent  call to the PullItemsAsync() method takes place and the remote table  should be updated.
The problem occurs only when debugging with Windows machine (platform windows) . When I use the Android emulator the logic performs as expected  without errors.

You may find the part of the Initialize method which constructs the offline store at the end of my answer.

(Being here  is there any link / article/ library to learn how to construct and use ITableQuery<U> queries  in IOfflineTable ?)

Thanks in advance!!!

```csharp
 // Create the offline store  

 private DatasyncClient _clientOff = null;
 private IOfflineTable<TodoItem> _tableoff = null;

 var connectionString = new UriBuilder { Scheme = "file", Path = OfflineDb, Query = "?mode=rwc" }.Uri.ToString();
 var store = new OfflineSQLiteStore(connectionString);
 store.DefineTable<TodoItem>();
 var optionsoff = new DatasyncClientOptions
 {
     OfflineStore = store,
     HttpPipeline = new HttpMessageHandler[] { new LoggingHandler() }
  };

 // Create the offline Datasync client.
 _clientOff = TokenRequestor == null
      ? new DatasyncClient(Constants.ServiceUri, optionsoff)
      : new DatasyncClient(Constants.ServiceUri, new GenericAuthenticationProvider(TokenRequestor), optionsoff);       

    await _clientOff.InitializeOfflineStoreAsync();
   _tableoff = _clientOff.GetOfflineTable<TodoItem>();
```

**Reply by adrianhall** [reply]
Just to clarify - this is for Xamarin.Forms with a windows target?  (That platform is not one we support, so I wanted to be sure).  If so, which windows target?  UWP, WinUI, WPF?

**Reply by geath** [reply]
This is for TodoApp.MAUI with a windows target (WinUI)

**Reply by adrianhall** [reply]
Got it - I haven't completed the work on the MAUI tutorial yet - there were lingering MAUI issues, so this isn't surprising.  I'm going to re-start the MAUI investigations this week now that 17.4 is out.
